test/test_common/utility.h

```diff
@@ -48,6 +48,10 @@ namespace Envoy {
 
 #if defined(__has_feature) && __has_feature(thread_sanitizer)
 #define TSAN_TIMEOUT_FACTOR 3
+#elif defined(__has_feature) && __has_feature(address_sanitizer)
+#define TSAN_TIMEOUT_FACTOR 3
+#elif defined(__has_feature) && __has_feature(memory_sanitizer)
+#define TSAN_TIMEOUT_FACTOR 3
```

Comment:
What's the overhead of the other sanitizers? I was under the impression tsan was much much more expensive than the others / considering we haven't had this issue crop up for the others until now

Reply:
Since the issue is not happening in other tests, also not happening in ext_pro_integration_test.cc before the tests added by the new timeout API change, let me back out this change first.    With that, this PR only extend the new API timeout to a number bigger enough also counting TSAN factor.